Tests/iaas/flavor-naming/flavor-form-server.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+"""Simple webserver that can be used to test the flavor-form.py CGI script."""
+# adapted from https://github.com/python/cpython/blob/3.12/Lib/http/server.py
+from http.server import CGIHTTPRequestHandler, BaseHTTPRequestHandler, ThreadingHTTPServer
+from http.server import _get_best_family, _url_collapse_path
+import argparse
+import contextlib
+import os
+import socket
+import sys
+
+
+class RequestHandler(CGIHTTPRequestHandler):
+
+    def is_cgi(self):
+        collapsed_path = _url_collapse_path(self.path)
+        if not collapsed_path.startswith('/cgi-bin/'):
+            return False
+        self.cgi_info = '', collapsed_path.removeprefix('/cgi-bin')
+        return True
+
+
+def test(HandlerClass=BaseHTTPRequestHandler,
+         ServerClass=ThreadingHTTPServer,
+         protocol="HTTP/1.0", port=8000, bind=None):
+    """Test the HTTP request handler class.
+
+    This runs an HTTP server on port 8000 (or the port argument).
+
+    """
+    ServerClass.address_family, addr = _get_best_family(bind, port)
+    HandlerClass.protocol_version = protocol
+    with ServerClass(addr, HandlerClass) as httpd:
+        host, port = httpd.socket.getsockname()[:2]
+        url_host = f'[{host}]' if ':' in host else host
+        print(f"The form is accessible at http://{url_host}:{port}/cgi-bin/flavor-form.py")
+        try:
+            httpd.serve_forever()
+        except KeyboardInterrupt:
+            print("\nKeyboard interrupt received, exiting.")
+            sys.exit(0)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-b', '--bind', metavar='ADDRESS',
+                        help='bind to this address '
+                             '(default: all interfaces)')
```

Comment:
One tiny suggestion that I missed to include yesterday (sorry!):

```suggestion
    parser.add_argument('-b', '--bind', metavar='ADDRESS',
                        default='127.0.0.1',
                        help='bind to this address '
                             '(default: %(default)s)')
```

This will make the string printed above actually usable in the default case:

```
The form is accessible at http://127.0.0.1:8000/cgi-bin/flavor-form.py
```
vs.

```
The form is accessible at http://0.0.0.0:8000/cgi-bin/flavor-form.py
```

Reply:
@martinmo The variant with 0.0.0.0 does indeed work (at least on my machine).

Reply:
@mbuechse ok, I checked again and now it works here, too – don't know what I did wrong the first time :shrug: but nevertheless, I would recommend using a local interface because it is the more secure default.